overlord/ifacestate/helpers.go

```diff
@@ -156,7 +174,8 @@ func (m *InterfaceManager) regenerateAllSecurityProfiles() error {
 		}
 	}
 
-	return nil
+	sk := interfaces.SystemKey()
+	return osutil.AtomicWriteFile(dirs.SnapSystemKeyFile, []byte(sk), 0644, 0)
```

Comment:
Can you please add a comment that explains why the system key is in a separate file.